numpy/ma/tests/test_core.py

```diff
@@ -4496,6 +4496,14 @@ def test_where_structured_masked(self):
         assert_equal(ma, expected)
         assert_equal(ma.mask, expected.mask)
 
+    def test_masked_invalid_error(self):
+        a = np.arange(5, dtype=object)
+        a[3] = np.PINF
+        a[2] = np.NaN
+        with pytest.raises(TypeError,
+                           match="not supported for the input types"):
+            np.ma.masked_invalid(a)
+
```

Comment:
We were looking at this in the triage meeting as well.  I guess the test is great (I honestly still need to parse it fully).
But we are missing an additional test for the successful path that was fixed I think:  Checking that the array was indeed modified in-place with `copy=False`?

Reply:
Yeah, thanks.  It is true that `masked_where` is tested, but it is also just nice to see the fix in action in the PR and we mainly have integration tests anyway.

A test too much cannot hurt :).

I find it odd that we consider `inf` invalid by default, but that is not a change.

Thanks @cmarmo, lets get this in!